app/javascript/packs/homePageFeed.jsx

```diff
@@ -43,6 +43,92 @@ function sendFeaturedArticleAnalyticsGA4(articleId) {
   })();
 }
 
+function feedConstruct(
+  pinnedItem,
+  imageItem,
+  feedItems,
+  bookmarkedFeedItems,
+  bookmarkClick,
+  currentUserId,
+  timeFrame,
+) {
+  const commonProps = {
+    bookmarkClick,
+  };
+
+  const feedStyle = JSON.parse(document.body.dataset.user).feed_style;
+
+  if (imageItem) {
+    sendFeaturedArticleGoogleAnalytics(imageItem.id);
+    sendFeaturedArticleAnalyticsGA4(imageItem.id);
+  }
+
+  return feedItems.map((item) => {
+    // billboard is an html string
+    if (typeof item === 'string') {
+      return (
+        <div
+          key={item.id}
+          // eslint-disable-next-line react/no-danger
+          dangerouslySetInnerHTML={{
+            __html: item,
+          }}
+        />
+      );
+    }
+
+    if (Array.isArray(item) && item[0].podcast) {
+      return <PodcastEpisodes key={item.id} episodes={item} />;
+    }
+
+    if (typeof item === 'object') {
+      // For "saveable" props, "!=" is used instead of "!==" to compare user_id
+      // and currentUserId because currentUserId is a String while user_id is an Integer
```

Comment:
I'm probably just missing it, but it doesn't seem like this comment currently applies to this block?

Reply:
Resolved in https://github.com/forem/forem/pull/19600/files